### ISSUE TITLE
dmg file of my app is damaged on other users computers

### ISSUE THREAD

**Issue by okadali** [electron-userland/electron-builder#7602]
* **Electron-Builder Version**: ^23.6.0

* **Node Version**: v16.17.1

* **Electron Version**: ^20.3.8
* **Electron Type (current, beta, nightly)**: current

* **Target**: macos

I'm trying to create an app using react and electron. I used codes `react-scripts build` and `electron-builder build --mac  --win -c.extraMetadata.main=build/electron.js --publish never` to package my app and i get a dmg file. I can add the dmg file to my applications and run it without any problem. But when i try to open it on another macos computer i get the "this app is damaged" error, i get the same error if i download the same dmg file and try to open it.

So basically my problem is i can use the dmg file i packaged using electron builder without any problem but if i upload the same dmg file and download again it stops working.

I checked if I'm missing any dependency file while sending the dmg file but that is not the case.

This is the current state of my `build` on package.json

I attached a picture of my `build` on package.json
![build](https://github.com/electron-userland/electron-builder/assets/72859603/6d96f729-05d3-4cda-8718-4e4c0d4e6a9e)

**Comment by okadali** [comment]
It's a problem about not signing the app

**Comment by AlexSeekube** [comment]
@okadali 
Hello,

So if I understand correctly, if I want to distribute my application to other OSX users, I must absolutely pay for the Apple Developer Program subscription?
Or is there another way to sign apps for free?

N.B: I don't want to put my application on the App Store.

**Comment by okadali** [comment]
@AlexSeekube 
Hi, 

Yes exactly like you said. If you want to distribute your application to other OSX users you have to sign it and to my knowledge there is no free alternative. It's nothing to do with puting your application to App Store, it's a process that required for OSX to trust your app.

**Comment by billmorgan123** [comment]
> @AlexSeekube Hi,
> 
> Yes exactly like you said. If you want to distribute your application to other OSX users you have to sign it and to my knowledge there is no free alternative. It's nothing to do with puting your application to App Store, it's a process that required for OSX to trust your app.

sir can you guid me how to make the "Developer ID Application" certificate but i require in p12 formate with key

**Comment by artashes-sanoyan** [comment]
For anyone having this issue you can try to [check this out](https://disable-gatekeeper.github.io/).

**Comment by aaort** [comment]
For anyone having this issue you can try to [check this out](https://disable-gatekeeper.github.io/).

**Comment by demensdeum** [comment]
There is quarantine (com.apple.quarantine ) flag that added for all downloaded dmg/zip files, all files inside are also quarantined.
You can check metadata flags by:
ls -l@ app.dmg

To remove flag, use xattr -d com.apple.quarantine app.dmg
After remove of com.apple.quarantine flag there will be no more "damaged app error"
